messenger.go

```diff
@@ -921,6 +910,47 @@ func Identicon(id string) (string, error) {
 	return identicon.GenerateBase64(id)
 }
 
+// VerifyENSName verifies that a registered ENS name matches the expected public key
+func (m *Messenger) VerifyENSNames(rpcEndpoint, contractAddress string, ensDetails []ens.ENSDetails) (map[string]ens.ENSResponse, error) {
+	verifier := ens.NewVerifier(m.logger)
+
+	ensResponse, err := verifier.CheckBatch(ensDetails, rpcEndpoint, contractAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update contacts
+	var contacts []Contact
+	for _, details := range ensResponse {
+		if details.Error == nil {
+			contact, err := buildContact(details.PublicKey)
+			if err != nil {
+				return nil, err
+			}
+			contact.ENSVerified = details.Verified
+			contact.ENSVerifiedAt = time.Now().Unix()
+			contact.Name = details.Name
+
+			contacts = append(contacts, *contact)
+		} else {
+			m.logger.Warn("Failed to resolve ens name",
+				zap.String("name", details.Name),
+				zap.String("publicKey", details.PublicKeyString),
+				zap.Error(details.Error),
+			)
+		}
+	}
+
+	if len(contacts) != 0 {
+		err = m.persistence.SetContactsENSData(contacts)
```

Comment:
```suggestion
		err := m.persistence.SetContactsENSData(contacts)
```

no need to leak it outside of the block.